pkg/core/blockchain.go

```diff
@@ -277,6 +277,7 @@ func (bc *Blockchain) AddHeaders(headers ...*Header) (err error) {
 		}
 	}
 	<-bc.headersOpDone
+	updateHeaderHeightMetric(bc.HeaderHeight())
```

Comment:
Actually, why not put it just under the `if oldlen != headerList.Len()`? This way you won't need to jump through the `headersOp` channel again.

Reply:
It should be something like `updateHeaderHeightMetric(headerList.Len())`, you're under a lock and can use it directly.